aten/src/ATen/native/xnnpack/Common.h

```diff
@@ -0,0 +1,78 @@
+#pragma once
+
+#include <ATen/ATen.h>
+
+#ifdef USE_XNNPACK
+
+#include <xnnpack.h>
+
+namespace at {
+namespace native {
+namespace xnnpack {
+namespace internal {
+
+struct Layout final {
+  // 4D Activation Maps
+  struct Activation4D final {
+    static constexpr size_t Batch = 0u;
+    static constexpr size_t Channels = 1u;
+    static constexpr size_t Height = 2u;
+    static constexpr size_t Width = 3u;
+  };
+
+  // ND Activation Maps
+  struct ActivationND final {
+    static int64_t Batch(const IntArrayRef tensor) {
+      if (C10_UNLIKELY(tensor.empty())) {
+        return -1;
+      }
+
+      int64_t batch = 1;
+
+      for (size_t index = 0u, dimension = tensor.size() - 1u; index < dimension; ++index) {
```

Comment:
nit: there's no point in defining `dimension` variable

Reply:
It's used in the loop condition.

Reply:
`for (size_t index = 0u, index < tensor.size()-1u; ++index)`

Reply:
Calling functions in a loop unnecessarily is not good practice in my opinion and I tend to avoid that habitually.   Here's someone who showed up in my twitter feed a while ago who thought it would be cheap unless it wasn't.  https://twitter.com/trav_downs/status/1216155531057995782

Reply:
Showed up in my feed too. Size is notorious to be called repeatedly.

Reply:
Even in the twitter thread you linked it is shown that a call to .size() is hoisted if vector is not modified inside the loop. In this case, `tensor` is const argument, so it cannot be modified, so call to `.size()` would be hoisted. However, even this is moot, because `tensor` is IntArrayRef, and a call to `.size()`, even if not hoisted,  is returning `Length` field, no divisions necessary. Let's not let premature microoptimizations make us write unreadable code. As the linked twitter thread correctly states, compilers exist for a reason.

Reply:
The unreliability of this code is a matter of your opinion.  It certainly does not look unreadable to me.